.github/workflows/reusable_ragger_tests_latest_speculos.yml

```diff
@@ -82,7 +83,7 @@ jobs:
       uses: actions/checkout@v4
       with:
           repository: ${{ inputs.app_repository }}
-          ref: ${{ inputs.app_branch_name }}
+          ref: ${{ matrix.device == 'nanos' && 'nanos_baseline' || inputs.app_branch_name  }}
```

Comment:
Don't harcode specific boilerplate data here, this job is a reusable generic job, we need to modify the caller